### PR TITLE
sl-50: User avatar

### PR DESCRIPTION
**Pull request by dimapopovych**
image

### CODE REVIEW

File: frontend/src/slices/auth/auth.slice.ts [hunk 1]
```diff
@@ -22,23 +22,38 @@ const { reducer, actions, name } = createSlice({
...
-      state.user = action.payload;
-      state.currentUserDataStatus = DataStatus.FULFILLED;
-    });
-    builder.addCase(usersActions.updateUser.fulfilled, (state, action) => {
-      state.user = action.payload;
+    builder.addCase(getCurrentUser.pending, (state) => {
+      state.currentUserDataStatus = DataStatus.PENDING;
     });
+    builder.addMatcher(
+      isAnyOf(
```
Comment (what1s1ove) [inline frontend/src/slices/auth/auth.slice.ts]:
```suggestion
```
avoid this for now. we will have a different approach with data statuses. write all cases explicitly

File: frontend/src/pages/account-settings/account-settings.tsx [hunk 2]
```diff
@@ -28,8 +28,26 @@ const AccountSettings: React.FC = () => {
     [dispatch],
   );
 
+  const handleUpdateUserAvatar = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      const avatar = event.target.files?.item(0);
```
Comment (what1s1ove) [inline frontend/src/pages/account-settings/account-settings.tsx]:
```suggestion
      const [avatar] = event.target.files ?? [];
```

File: frontend/src/pages/account-settings/account-settings.tsx [hunk 3]
```diff
@@ -28,8 +28,26 @@ const AccountSettings: React.FC = () => {
...
+
+      if (avatar) {
+        const formData = new FormData();
+        formData.append('file', avatar);
```
Comment (what1s1ove) [inline frontend/src/pages/account-settings/account-settings.tsx]: add enum for `file` and use for both app (back & front)

File: backend/src/packages/users/user.repository.ts [hunk 4]
```diff
@@ -5,16 +5,18 @@ import { type UserModel } from '~/packages/users/user.model.js';
 class UserRepository implements Omit<IRepository, 'delete'> {
   private userModel: typeof UserModel;
 
+  private defaultRelationExpression = 'userDetails.avatar';
```
Comment (what1s1ove) [inline backend/src/packages/users/user.repository.ts:8]:
```suggestion
```
write them explicitly

File: backend/src/libs/packages/server-application/server-app.ts [hunk 5]
```diff
@@ -13,7 +14,7 @@ import { type IDatabase } from '~/libs/packages/database/database.js';
 import { HttpCode, HttpError } from '~/libs/packages/http/http.js';
 import { type ILogger } from '~/libs/packages/logger/logger.js';
 import { token } from '~/libs/packages/token/token.js';
-import { authorization } from '~/libs/plugins/authorization/authorization.plugin.js';
+import { authorization, file } from '~/libs/plugins/plugins.js';
```
Comment (what1s1ove) [inline backend/src/libs/packages/server-application/server-app.ts]:
```suggestion
import { authorization as authorizationPlugin, file as filePlugin } from '~/libs/plugins/plugins.js';
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: backend/src/packages/users/user.repository.ts (hunk 4 above)
Comment (dimapopovych) [reply backend/src/packages/users/user.repository.ts:8]: linter forces to declare a constant

Comment (what1s1ove) [reply backend/src/packages/users/user.repository.ts:8]: okay, expected then

File: backend/src/packages/users/user.controller.ts [hunk 6]
```diff
@@ -67,32 +71,6 @@ class UserController extends Controller {
           }>,
         ),
     });
-
-    this.addRoute({
```
Comment (what1s1ove) [inline backend/src/packages/users/user.controller.ts]: return them back

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**